### DISCUSSION TITLE
N-fold Cartesian Product

### DISCUSSION THREAD

**Discussion by fweth** [josdejong/mathjs#2688] [Ideas]
Hello, it would be really cool if `setCartesian` would support more than two arguments.  Alternatively, if the convert-multi-dimension-arrays feature could be disabled, possibly by a boolean, one could do something like `map(setCartesian(setCartesian([1,2,3], [4,5,6]), [7,8,9]), flatten)`.

**Comment by josdejong** [comment]
Ah, that is a good idea indeed. Shouldn't be to hard implement.

Anyone interested in extending `setCarthesian` to support more than two arguments?
